aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -1215,23 +1125,17 @@ module aptos_framework::coin {
         if (is_account_registered<CoinType>(account_addr)) { return };
 
         account::register_coin<CoinType>(account_addr);
-        let coin_store = CoinStore<CoinType> {
-            coin: Coin { value: 0 },
-            frozen: false,
-            deposit_events: account::new_event_handle<DepositEvent>(account),
-            withdraw_events: account::new_event_handle<WithdrawEvent>(account)
-        };
-        move_to(account, coin_store);
```

Comment:
Does this mean there could be some coin stores that you don't expect?

Also, shouldn't you just make this whole function a no-op after checking for permission.  Doesn't really matter if it's registered and make that call.

Oh, nevermind it checks if the coin exists there